src/Microsoft.OData.Client/ALinq/GroupByAnalyzer.cs

```diff
@@ -0,0 +1,529 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="GroupByAnalyzer.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+namespace Microsoft.OData.Client
+{
+    #region Namespaces
+
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Linq.Expressions;
+    using System.Reflection;
+    using Microsoft.OData.UriParser.Aggregation;
+
+    #endregion Namespaces
+
+    /// <summary>
+    /// Analyzes an expression to check whether it can be satisfied with $apply.

```

Comment:
is it with $apply or $groupby? can we use $apply without groupby?

Reply:
@ElizabethOkerio We can use `$apply` without `groupby`. `$apply` is the query option while `groupby` (no leading `$`) is a transformation. Other supported transformations are `aggregate` and `filter`, i.e.:
- `$apply=groupby(...)`
- `$apply=aggregate(...)`
- `$apply=filter(...)`